src/lib/components/vegachart/VegaChart.component.js

```diff
@@ -60,21 +61,36 @@ function VegaChart({ id, spec, theme = "light" }: Props) {
   };
   const themedSpec = { ...spec, ...themeConfig };
 
+  const vegaInstance = useRef<Result>();
+
   useEffect(() => {
-    vegaEmbed(`#${id}`, themedSpec, {
-      renderer: "svg",
-      tooltip: { theme: theme },
-      // Determines if action links
-      // ("Export as PNG/SVG", "View Source", "View Vega" (only for Vega-Lite), "Open in Vega Editor")
-      // are included with the embedded view.
-      // If the value is true, all action links will be shown and none if the value is false.
-      actions: false,
-    })
-      .then(function (result) {
-        // Access the Vega view instance (https://vega.github.io/vega/docs/api/view/) as result.view
+      let isMounted = true;
+      
+      vegaEmbed(`#${id}`, themedSpec, {
+        renderer: "svg",
+        tooltip: { theme: theme },
+        // Determines if action links
+        // ("Export as PNG/SVG", "View Source", "View Vega" (only for Vega-Lite), "Open in Vega Editor")
+        // are included with the embedded view.
+        // If the value is true, all action links will be shown and none if the value is false.
+        actions: false,
+      })
+      .then(result => {
+        vegaInstance.current = result;
       })
-      .catch(console.error);
-  }, [id, themedSpec, theme]);
+      .catch((...args) => {
+        if (isMounted) {
+          console.error(...args) // TODO: we should handle this with a retry or an error state of the component
+        }
+      });
+      
+      return () => {
+        isMounted = false;
+        if (vegaInstance.current) {
```

Comment:
You might also want to make sure that the view property exists.

Reply:
Thanks for sharing this however according to the typing of vega Result (https://github.com/vega/vega-embed/blob/master/src/embed.ts#L119) and the implementation view will always be defined on this object so I think we can safely skip this check thanks to flow/ts typings